### PR TITLE
Route action without ns

### PR DESCRIPTION
**Pull request by arianaw66**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2730

### CODE REVIEW

**Comment by arianaw66** [comment]
Screen Shot 2020-05-01 at 1 28 39 PM

**Comment by arianaw66** [comment]
Still need to add unit tests and fix the routeAction multi case.

**Comment by ashleywang1** [comment]
Are we updating solo-kit in this PR to pick up the change there as well?

**Comment by arianaw66** [comment]
No, with this PR in place, the solo-kit change shouldn't affect this issue any longer. I plan to open a separate PR to fix it, though, since it is a bug we discovered while working on this issue.

**Comment by arianaw66** [comment]
Docker-login fail. Flake? /kick

**Comment by arianaw66** [comment]
regression test /kick

File: projects/gateway/pkg/translator/converter.go [hunk 1]
```diff
@@ -221,6 +221,20 @@ func (rv *routeVisitor) visit(resource resourceWithRoutes, parentRoute *routeInf
...
 
+			// if this is a routeAction pointing to an upstream without specifying the namespace, set the namespace to that of the parent resource
+			if action, ok := routeClone.Action.(*gatewayv1.Route_RouteAction); ok {
+				parentNamespace := resource.InputResource().GetMetadata().Namespace
+				if upstream := action.RouteAction.GetSingle().GetUpstream(); upstream != nil && upstream.GetNamespace() == "" {
+					upstream.Namespace = parentNamespace
+				}
+				if multiDests := action.RouteAction.GetMulti().GetDestinations(); multiDests != nil {
+					for _, dest := range multiDests {
+						if upstream := dest.GetDestination().GetUpstream(); upstream != nil && upstream.GetNamespace() == "" {
```
Comment (ashleywang1) [inline projects/gateway/pkg/translator/converter.go:232]: nit: move the comment "// if this is a routeAction pointing to an upstream without specifying the namespace, set the namespace to that of the parent resource" here instead?

File: changelog/v1.4.0-beta9/fix-routeAction-without-ns.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+changelog:
+  - type: FIX
+    description: >
+      If a virtual service or route table has a route using a routeAction that points to an upstream but does not specify the namespace where the upstream is, Gloo will assume it's in the namespace of the parent resource (the upstream group or the virtual service or route table itself). Previously, this caused Envoy to be in a bad state, and the route would return a 503 error.
```
Comment (ashleywang1) [inline changelog/v1.4.0-beta9/fix-routeAction-without-ns.yaml:4]:
```suggestion
      We allow virtual services, route tables, and upstreams to refer to an upstream by name only, leaving the namespace field empty. Gloo will assume that the upstream's namespace is the namespace of the parent resource (the upstream group, virtual service or route table). Previously, this caused Envoy to be in a bad state, and the route would return a 503 error.
```

**Pull request merged**